### PR TITLE
Docs, follow best practices

### PR DESCRIPTION
**Pull request by svx**
This follow our best practices 

- Update README, according to styleguide
- Fix rst in README
- Add docs/index.rst
- Fix typos
- Fix rst [link syntax]
- Fix rst [code-blocks]
- Move non docs related files out of /docs [following best practices]

### CODE REVIEW

**Comment by svx** [comment]
I do not get this the rst of the README is valid for PyPi
rst-lint-valid

**Comment by do3cc** [comment]
Did you look at this: https://github.com/plone/bobtemplates.plone/blob/8c7d14dd331e530dc704c2724e0c2e3a25c858da/buildout.cfg#L76-L82

**Comment by mauritsvanrees** [comment]
Travis fails with:

```
$ bin/check-readme
running check
warning: check: Could not finish the parsing.
error: Please correct your package.
```

That leads to this issue: http://bugs.python.org/issue23063
Solved here: https://hg.python.org/cpython/rev/38826e21f0db/
which says: 'Issue #23063: In the disutils' check command, fix parsing of reST with code or code-block directives.'
The fix is included since Python 2.7.10. So I guess the Python version on Travis is older...
Locally, when I install docutils, it works, but my Python is more recent.

File: README.rst [hunk 1]
```diff
@@ -95,16 +63,23 @@ Theme
...
+Documentation
+-------------
+
+Full documentation for end users can be found in the "docs" folder.
+
+It is also available online at http://docs.plone.org/develop/addons/bobtemplates.plone
+
 Installation
-============
+-------------
```
Comment (mauritsvanrees) [inline README.rst]: This underline for `Installation` is one dash too long, though I guess this does not lead to errors.

**Comment by svx** [comment]
OK, so I have to wait till travis updates python or find another way to write the readme, not sure yet what would be a good solution :)

Thanks @do3cc and @mauritsvanrees for looking into it !

**Comment by svx** [comment]
Dirty, but it works :)

**Comment by do3cc** [comment]
is code_block not sphinx specific anyway?

**Comment by svx** [comment]
@do3cc 
```
.. code:: bash
```

or 

```
.. code-block:: bash
```

works with rst in common, not only with sphinx :)

I use the latter for other readme's [rst only, no sphinx] on pypi and it works just fine :)

**Comment by do3cc** [comment]
I consider you the authority here ;-)

**Comment by svx** [comment]
@do3cc lol, here an example:
https://pypi.python.org/pypi/sphinxcontrib-osexample/0.1.1

**Comment by mauritsvanrees** [comment]
You have moved the license files out of the docs directory. The [styleguide](http://docs.plone.org/about/documentation_styleguide.html) says otherwise: "Include a /docs/LICENSE.rst with a short description of the license, and /docs/LICENSE.GPL for the legalese."
Does the styleguide need updating?

**Comment by mauritsvanrees** [comment]
Otherwise: +1

**Pull request merged**